src/main/java/tool/designpatterns/verifiers/multiclassverifiers/AdapterVerifier.java

```diff
@@ -0,0 +1,238 @@
+            for (AnnotationExpr annotation : method.getAnnotations()) {
+                if (annotation.getNameAsString().equalsIgnoreCase("override")) {
+                    resultList.add(isWrapper(method, currentClass, adaptee));
+                    return resultList;
+                }
+            }
+            resultList.add(Boolean.FALSE);
+            return resultList;
+        }
+
+        /**
+         * A method that checks if a method is called from within another method, i.e. if it is
+         * wrapped.
+         *
+         * @param method  The MethodDeclaration fo the wrapping method.
+         * @param adaptee The ClassOrInterfaceDeclaration to look for the wrapped method in.
+         * @return a boolean, true if it is wrapped, otherwise false.
+         */
+        private Boolean isWrapper(
+            MethodDeclaration method, ClassOrInterfaceDeclaration currentClass,
+            ClassOrInterfaceDeclaration adaptee) {
+            List<Boolean> list = method.accept(new MethodCallVisitor(currentClass), adaptee);
+
+            return list.stream().anyMatch(e -> e);
+        }
+    }
+
+
+    /**
+     * A class used to visit every method call expression node in a ClassOrInterfaceDeclaration.
+     */
+    private static class MethodCallVisitor
+        extends GenericListVisitorAdapter<Boolean, ClassOrInterfaceDeclaration> {
+
+        private final ClassOrInterfaceDeclaration currentClass;
+
+        public MethodCallVisitor(ClassOrInterfaceDeclaration currentClass) {
+            super();
+            this.currentClass = currentClass;
+        }
+
+        /**
+         * Checks if the adaptee is an interface or a superclass, and checks whether the method call
+         * wraps a method call from the adaptee, if it does it adds true to the list otherwise it
+         * returns false.
+         *
+         * @param methodCallExpr the type of node to be visited
+         * @param adaptee        the ClassOrInterfaceDeclaration nodes are being visited in.
+         * @return a list of Booleans, true if wrapped method call otherwise false.
+         */
+        @Override
+        public List<Boolean> visit(
+            MethodCallExpr methodCallExpr, ClassOrInterfaceDeclaration adaptee) {
+            List<Boolean> boolList = super.visit(methodCallExpr, adaptee);
+
+            if (adaptee.isInterface()) {
+                for (FieldDeclaration field : currentClass.getFields()) {
+                    if (field.getCommonType().toString().equals(adaptee.getNameAsString())) {
+                        boolList.add(Boolean.TRUE);
+                        return boolList;
+                    }
+                }
+            } else if (methodCallExpr.getScope().get().toString().equalsIgnoreCase("super")) {
+                boolList.add(Boolean.TRUE);
+                return boolList;
+            }
+
+            boolList.add(Boolean.FALSE);
+            return boolList;
+        }
+    }
```

Comment:
Consider breaking out to a separate (perhaps package private) class.

Reply:
I prefer this to be an inner class to not clutter the file structure. I do not see a good way to abstract it to be usable in other verifiers. I leave this conversation up if someone else has strong opinions on the subject.